x-pack/test_serverless/api_integration/test_suites/common/security/authentication.ts

```diff
@@ -0,0 +1,186 @@
+              .post('/api/security/oidc/initiate_login')
+              .set(svlCommonApi.getInternalRequestHeader());
+            svlCommonApi.assertApiNotFound(body, status);
+          });
+
+          it('OIDC login (deprecated)', async () => {
+            const { body, status } = await supertest
+              .post('/api/security/v1/oidc')
+              .set(svlCommonApi.getInternalRequestHeader());
+            svlCommonApi.assertApiNotFound(body, status);
+          });
+
+          it('OIDC 3rd party login', async () => {
+            const { body, status } = await supertest
+              .get('/api/security/oidc/initiate_login')
+              .set(svlCommonApi.getInternalRequestHeader());
+            svlCommonApi.assertApiNotFound(body, status);
+          });
+        });
+
+        it('SAML callback (deprecated)', async () => {
+          const { body, status } = await supertest
+            .post('/api/security/v1/saml')
+            .set(svlCommonApi.getInternalRequestHeader());
+          svlCommonApi.assertApiNotFound(body, status);
+        });
+      });
+
+      describe('internal', () => {
+        it('get current user', async () => {
+          let body: any;
+          let status: number;
+
+          ({ body, status } = await supertest
+            .get('/internal/security/me')
+            .set(svlCommonApi.getCommonRequestHeader()));
+          // expect a rejection because we're not using the internal header
+          expect(body).toEqual({
+            statusCode: 400,
+            error: 'Bad Request',
+            message: expect.stringContaining(
+              'method [get] exists but is not available with the current configuration'
+            ),
+          });
+          expect(status).toBe(400);
+
+          ({ body, status } = await supertest
+            .get('/internal/security/me')
+            .set(svlCommonApi.getInternalRequestHeader()));
+          // expect success because we're using the internal header
+          expect(body).toEqual({
+            authentication_provider: { name: '__http__', type: 'http' },
+            authentication_realm: { name: 'reserved', type: 'reserved' },
+            authentication_type: 'realm',
+            elastic_cloud_user: false,
+            email: null,
+            enabled: true,
+            full_name: null,
+            lookup_realm: { name: 'reserved', type: 'reserved' },
+            metadata: { _reserved: true },
+            roles: ['superuser'],
+            username: 'elastic',
+          });
+          expect(status).toBe(200);
+        });
+      });
+
+      describe('public', () => {
+        it('logout', async () => {
+          const { body, status } = await supertest
+            .get('/api/security/logout')
+            .set(svlCommonApi.getCommonRequestHeader());
+
+          // Should fail with 400 (not 404) because it cannot process redirect (but for no other reason)
+          // Can we set this up to mock a successsful redirect?
```

Comment:
> Can we set this up to mock a successsful redirect?

I think if you omit `.set(svlCommonApi.getCommonRequestHeader())` it should do it (to not send `kbn-xsrf` header).